Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -34,11 +34,62 @@ public extension Array {
         swapAt(index, otherIndex)
     }
 }
+// MARK: - Methods (Hashable)
+public extension Array where Element: Hashable {
+    /// SwifterSwift: Remove all instances contained in items parameter from array. Complexity: O(n)
+    ///
+    ///        [1, 2, 2, 3, 4, 5].removeAll([2,5]) -> [1, 3, 4]
+    ///        ["h", "e", "l", "l", "o"].removeAll(["l", "h"]) -> ["e", "o"]
+    ///
+    /// - Parameter items: unique items to remove.
+    /// - Returns: self after removing all instances of all items in given array.
+    @discardableResult
+    mutating func removeAll(_ items: Set<Element>) -> [Element] {
+        guard !items.isEmpty else { return self }
+        removeAll(where: { items.contains($0) })
+        return self
+    }
+    
+    /// SwifterSwift: Remove all duplicate elements from Array, maintaining order. Complexity: O(n)
+    ///
+    ///        [1, 2, 2, 3, 4, 5].removeDuplicates() -> [1, 2, 3, 4, 5]
+    ///        ["h", "e", "l", "l", "o"]. removeDuplicates() -> ["h", "e", "l", "o"]
+    ///
+    /// - Returns: Return array with all duplicate elements removed.
+    @discardableResult
+    mutating func removeDuplicates() -> [Element] {
+        var foundElements = Set<Element>()
+        self = reduce(into: [Element]()) {
```

Comment:
I think that could be better if we could use `removeAll(where: )` ... that way we avoid assign to self and creating a new array... taking advantage on the already performant partion based in-place mutation. 
 what do you think?

Reply:
How would you remove "all but the first"?

Reply:
We already have a similar implementation in `RangeReplaceableCollectionExtensions.swift`.
```Swift
mutating func removeDuplicates() {
    var set = Set<Element>()
    removeAll { !set.insert($0).inserted }
}
```